codalab/lib/upload_manager.py

```diff
@@ -258,3 +326,168 @@ def cleanup_existing_contents(self, bundle):
         bundle_update = {'data_hash': None, 'metadata': {'data_size': 0}}
         self._bundle_model.update_bundle(bundle, bundle_update)
         self._bundle_model.update_user_disk_used(bundle.owner_id)
+
+    def get_bundle_sas_token(self, path, **kwargs):
+        """
+        Get SAS token with write permission. Used for bypass server uploading.
+        """
+        return (
+            parse_linked_bundle_url(path)
+            .bundle_path_sas_url(permission='rw', **kwargs)
```

Comment:
Should this be

```suggestion
            .bundle_path_sas_url(permission='w', **kwargs)
```

Reply:
> Should this be

This token is used for upload first, then used for download this file to generate index file. So the permission is `read and write`